store/displayPreferences.ts

```diff
@@ -134,170 +98,119 @@ export const actions: ActionTree<
     } catch (error) {
-      await dispatch('initStateFailure', { error });
+      const message = this.$i18n.t('failedRetrievingDisplayPreferences');
+      await dispatch(
+        'snackbar/pushSnackbarMessage',
+        { message, color: 'error' },
+        { root: true }
+      );
     }
   },
-
-  /**
-   * On query success, stores the result and call the callbacks
-   *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.commit - Vuex commit
-   * @param {any} param0.dispatch - Vuex dispatch
-   * @param {any} param1 - Payload
-   * @param {DisplayPreferencesDto} param1.displayPreferences - Display preferences object
-   */
-  async initStateSuccess(
-    { commit, dispatch },
-    { displayPreferences }: { displayPreferences: DisplayPreferencesDto }
-  ) {
-    commit('INIT_STATE', { displayPreferences });
-    await dispatch('callAllCallbacks');
-  },
-
-  /**
-   * On query error, sends the error and message to the store logger
-   *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.dispatch - Vuex dispatch
-   * @param {any} param1 - Payload
-   * @param {any} param1.error - Try-catch error
-   */
-  async initStateFailure({ dispatch }, { error }: { error: unknown }) {
-    const message = this.$i18n.t('failedRetrievingDisplayPreferences');
-    await dispatch('requestError', { message, error });
-  },
-
   /**
    * Pushes the current state to the server
    *
-   * @param {any} param0 - Vuex
-   * @param {any} param0.state - Vuex state
-   * @param {any} param0.dispatch - Vuex dispatch
+   * @param {any} context - Vuex action context
+   * @param {any} context.state - Vuex state
+   * @param {any} context.dispatch - Vuex dispatch
    */
-  async pushState({ state, dispatch }) {
+  async pushState({ state, dispatch, commit }) {
+    commit('SYNCING_STARTED');
     try {
+      // The fetch part is done because DisplayPreferences doesn't accept partial updates
+      const responseFetch = await this.$api.displayPreferences.getDisplayPreferences(
+        {
+          displayPreferencesId: 'usersettings',
+          userId: this.$auth.user?.Id,
+          client: 'vue'
+        }
+      );
+
+      if (responseFetch.status !== 200)
+        throw new Error(
+          'get display preferences status response = ' + responseFetch.status
+        );
+
+      const displayPrefs = responseFetch.data;
+      displayPrefs.CustomPrefs = {};
+      Object.assign(displayPrefs.CustomPrefs, state);
+      for (const [key, value] of Object.entries(displayPrefs.CustomPrefs)) {
+        /**
+         * As TypeScript typings are erased at runtime, we can't rely on typings to distinguish between
+         * the variables that belong just to the store and the ones that we want to be synced to the server.
+         *
+         * We can overcome this limitation in the future by storing our prefs in another store and leaving this store
+         * just for the API calls and the state of syncing.
+         */
```

Comment:
Opinions on doing this in a follow-up PR to simplify this stuff? @Camc314 @ThibaultNocchi

Reply:
I'm also leaning to this approach so we can also handle diffs between local settings and remote settings better.

The idea basically is to have another store that stores the client settings so everytime a mutation happen in that store, this copies it and push to server.

Right now for instance the settings from server always take precedence. However, if settings are changed in another Vue while offline, client settings will have precedence over server settings, so I expect we should store the last synced date alongside ``syncing``, which means adding another "exception" to the for loop where we prepare the data to push to server.

Reply:
Yeah makes sense. IMO, we should have our client store and the displaypreferences mixin (or any other mixin, not a store though) handle the syncing. WDYT?

Reply:
Not sure about the move to the mixin part as if we want to watch the syncing status and the date synced states we can't do it in a mixin in a clear way.

Reply:
Yeah that's true. We'll think about when the time comes :)